client/src/state/historyviewer/HistoryViewerReducer.js

```diff
@@ -0,0 +1,27 @@
+import HISTORY_VIEWER from './HistoryViewerActionTypes';
+
+const initialState = {
+  currentVersion: 0,
+  loading: false,
+};
+
+/**
+ * Reducer for the `versionedAdmin.historyViewer` state key.
+ *
+ * @param {object} state
+ * @param {string} type - Name of the dispatched action.
+ * @param {object} payload - Optional data passed with the action.
+ */
+export default function historyViewerReducer(state = initialState, { type, payload } = {}) {
+  switch (type) {
+    case HISTORY_VIEWER.SET_CURRENT_VERSION: {
+      return {
+        ...state,
+        currentVersion: payload.id,
+      };
+    }
```

Comment:
This is good, but the store persists its state after the component is unmounted, meaning when I go from the Home history view, select version 1, then to the About Us history view, Version 1 is already selected. It also means that when I go to a page that doesn't have version 1, it throws, and I have to refresh the browser.

This is an a atypical scenario, because the history viewer is an ephemeral component in a persistent environment, so all of the garbage collection has to be imperative. In a proper SPA, we don't have to worry about this.

A `componentWillUnmount()` handler that calls a reset action in one of the top level components should suffice.